src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1

```diff
@@ -0,0 +1,177 @@
+
+<#
+	.SYNOPSIS
+	Tests creating a stretch database on V12.0 server
+#>
+function Test-CreateStretchDatabase
+{
+	Test-CreateDatabaseInternal "12.0" "Southeast Asia"
+}
+
+<#
+	.SYNOPSIS
+	Helper for testing creating a database
+#>
+function Test-CreateDatabaseInternal ($serverVersion, $location = "Japan East")
+{
+	# Setup
+	$rg = Create-ResourceGroupForTest
+	$server = Create-ServerForTest $rg $serverVersion $location
+
+	try
+	{
+		# Create stretch database with all parameters
+		$databaseName = Get-DatabaseName
+		$collationName = "SQL_Latin1_General_CP1_CI_AS"
+		$maxSizeBytes = 250GB
+		$strechdb = New-AzureRmSqlDatabase -ResourceGroupName $rg.ResourceGroupName -ServerName $server.ServerName -DatabaseName $databaseName `
+				-CollationName $collationName -MaxSizeBytes $maxSizeBytes -Edition Stretch -RequestedServiceObjectiveName DS100
+		Assert-AreEqual $databaseName $strechdb.DatabaseName 
+		Assert-AreEqual $maxSizeBytes $strechdb.MaxSizeBytes 
+		Assert-AreEqual Stretch $strechdb.Edition 
+		Assert-AreEqual DS100 $strechdb.CurrentServiceObjectiveName
+		Assert-AreEqual $collationName $strechdb.CollationName 
+	}
+	finally
+	{
+		Remove-ResourceGroupForTest $rg
+	}
+}
+
+<#
+	.SYNOPSIS
+	Tests updating a stretch database on V12.0 server
+#>
+function Test-UpdateStretchDatabase
+{
+	Test-UpdateDatabaseInternal "12.0" "Southeast Asia"
+}
+
+<#
+	.SYNOPSIS
+	Helper for testing updating a database
+#>
+function Test-UpdateDatabaseInternal ($serverVersion, $location = "Japan East")
+{
+	# Setup
+		$rg = Create-ResourceGroupForTest
+		$server = Create-ServerForTest $rg $serverVersion $location
+	try {
+		# Create stretch database
+		$databaseName = Get-DatabaseName
+		$collationName = "SQL_Latin1_General_CP1_CI_AS"
+		$maxSizeBytes = 250GB
+		$strechdb = New-AzureRmSqlDatabase -ResourceGroupName $rg.ResourceGroupName -ServerName $server.ServerName -DatabaseName $databaseName `
+		-CollationName $collationName -MaxSizeBytes $maxSizeBytes -Edition Stretch -RequestedServiceObjectiveName DS100
+
+		# Alter stretch database properties
+		$strechdb2 = Set-AzureRmSqlDatabase -ResourceGroupName $rg.ResourceGroupName -ServerName $server.ServerName -DatabaseName $strechdb.DatabaseName `
+		-MaxSizeBytes $maxSizeBytes -Edition Stretch -RequestedServiceObjectiveName DS200
+		Assert-AreEqual $strechdb.DatabaseName $strechdb2.DatabaseName
+		Assert-AreEqual $maxSizeBytes $strechdb2.MaxSizeBytes
+		Assert-AreEqual Stretch $strechdb2.Edition
+		Assert-AreEqual DS200 $strechdb2.CurrentServiceObjectiveName
+		Assert-AreEqual $collationName $strechdb2.CollationName
+	}
+	finally
+	{
+		Remove-ResourceGroupForTest $rg
+	}
+}
+
+
+<#
+	.SYNOPSIS
+	Tests Getting a stretch database on V12.0 server
+#>
+function Test-GetStretchDatabase
+{
+	Test-GetDatabaseInternal "12.0" "Southeast Asia"
+}
+
+<#
+	.SYNOPSIS
+	Helper for testing getting a database
+#>
+function Test-GetDatabaseInternal  ($serverVersion, $location = "Japan East")
```

Comment:
I understand we can use existing Cmdlets to retrieve location list, but then how to select which one is the one that can satisfy every one?

Also, I just use the way how checked-in tests are testing so do i need to make this change to my code?